src/Hamiltonians/Molecular.jl

```diff
@@ -0,0 +1,780 @@
+)
+    ext = Base.get_extension(@__MODULE__, :ElemCoExt)
+    if isnothing(ext)
+        error("MolecularHamiltonian requires that ElemCo is loaded, i.e. `using ElemCo`")
+    end
+    fd = ext.read_fcidump(fcidump_path, Val(4))
+    return MolecularHamiltonian(fd, starting_address, fcidump_path)
+end
+
+function Base.show(io::IO, h::MolecularHamiltonian)
+    io = IOContext(io, :compact => true)
+    print(io, "MolecularHamiltonian(")
+    if ispath(h.specifier)
+        print(io, "\"", h.specifier, "\"")
+    else
+        print(io, typeof(h.fcidump))
+    end
+    print(io, ", starting_addresss=")
+    show(io, h.starting_address)
+    if ispath(h.specifier)
+        print(io, ")")
+    else
+        print(io, ", specifier=\"", h.specifier, "\")")
+    end
+end
+
+LOStructure(::Type{<:MolecularHamiltonian{<:Real}}) = IsHermitian()
+
+struct MolecularHamiltonianOperatorColumn{
+    A<:FermiFS2C,T,O<:MolecularHamiltonian{T,A},M<:FermiFS2CModes
+} <: AbstractOperatorColumn{A,T,O}
+    address::A # Contains address Psi_i, provided by operator_column
+    op::O   # Represent Hamiltonian itself, provided by operator_column
+    diag::T # < Psi_i | Psi_i >, calculated by diagonal_element
+    modes::M # Store the modes of address
+end
+
+function starting_address(h::MolecularHamiltonian)
+    return h.starting_address
+end
+
+function operator_column(
+    h::MolecularHamiltonian{T,A,D}, a::A
+)::MolecularHamiltonianOperatorColumn where {T<:Number,A<:FermiFS2C,D}
+    modes = full_mode_maps(a)
+
+    diag = zero(T)
+    diag_one_elec_int = one_electron_integral(h.fcidump.int1, modes.occupied)
+    diag_two_elec_int = two_electron_integral(h.fcidump.int2, modes.occupied)
+    diag = h.fcidump.int0 + diag_one_elec_int + diag_two_elec_int
+
+    return MolecularHamiltonianOperatorColumn{A,T,typeof(h),typeof(modes)}(
+        a, h, diag, modes
+    )
+end
+
+parent_operator(c::MolecularHamiltonianOperatorColumn) = c.op
+starting_address(c::MolecularHamiltonianOperatorColumn) = c.address
+
+function diagonal_element(
+    column::MolecularHamiltonianOperatorColumn{A,T,O,OD}
+) where {A<:FermiFS2C,T,O,OD}
+    return column.diag
+end
+
+function num_offdiagonals(column::MolecularHamiltonianOperatorColumn)
+    return length(offdiagonals(column))
+end
+
+function offdiagonals(column::MolecularHamiltonianOperatorColumn)
+    return MolecularHamiltonianOffDiagonals(column)
+end
+
+function random_offdiagonal(column::MolecularHamiltonianOperatorColumn)
+    ods = offdiagonals(column)
+    l = length(ods)
+    ri = rand(1:l)
+    state = linear_to_state(ods, ri)
+    r = iterate(ods, state)
+    if isnothing(r)
+        return (column.address, 0, 0)
+    end
+    return r[1][1], 1 / l, r[1][2]
+end
+
+"""
+    one_electron_integral(
+        int1::Array{T,2},
+        occ_modes::Tuple{AbstractVector{FermiFSIndex},AbstractVector{FermiFSIndex}}
+    )::T where {T<:Number}
+
+Calculate the one body operator diagonal term ``\\langle U | \\hat{H}_1 | U \\rangle``.
```

Comment:
```suggestion
    one_electron_diagonal(
        int1::Matrix{<:Number},
        occ_modes::NTuple{2,<:AbstractVector{FermiFSIndex}}
    )

Calculate the one body operator diagonal term ``⟨U|Ĥ₁|U⟩`` for a two-component Fermi Fock address.
```
I suggest to rename the function to make it clearer what the function does.